projects/plugins/jetpack/extensions/blocks/contact-form/components/jetpack-field-controls.js

```diff
@@ -18,13 +21,86 @@ import JetpackFieldWidth from './jetpack-field-width';
 import JetpackManageResponsesSettings from './jetpack-manage-responses-settings';
 
 const JetpackFieldControls = ( {
-	setAttributes,
-	width,
+	attributes,
 	id,
-	required,
 	placeholder,
 	placeholderField = 'placeholder',
+	required,
+	setAttributes,
+	width,
+	type,
 } ) => {
+	const setNumberAttribute = ( key, parse = parseInt ) => value => {
+		const parsedValue = parse( value, 10 );
+
+		setAttributes( {
+			[ key ]: ! isNaN( parsedValue ) ? parsedValue : '',
+		} );
+	};
+
+	let colorSettings = [];
```

Comment:
Here's a thought, what if we just defined all the settings here and then used the `hasBorderControl` variable (see below) to only pass the relevant ones to the component, avoiding duplication and unnecessary nesting. It'd look something like this:

```
const colorSettings = [
	{
		value: attributes.labelColor,
		onChange: value => setAttributes( { labelColor: value } ),
		label: __( 'Label text', 'jetpack' ),
	},
	{
		value: attributes.inputColor,
		onChange: value => setAttributes( { inputColor: value } ),
		label: __( 'Field text', 'jetpack' ),
	},
	{
		value: attributes.fieldBackgroundColor,
		onChange: value => setAttributes( { fieldBackgroundColor: value } ),
		label: __( 'Field Background', 'jetpack' ),
	},
	{
		value: attributes.borderColor,
		onChange: value => setAttributes( { borderColor: value } ),
		label: __( 'Field Border', 'jetpack' ),
	},
	{
		value: attributes.blockBackgroundColor,
		onChange: value => setAttributes( { blockBackgroundColor: value } ),
		label: __( 'Block Background', 'jetpack' ),
	},
];

...

<PanelColorSettings
	...
	colorSettings={ hasBorderControls ? colorSettings.slice( 0, 2 ) : colorSettings }
>
```

Reply:
Nice. That should work. Thanks. I'll give that a go.